mos-platform/pet/link.ld

```diff
@@ -0,0 +1,19 @@
+/* Commodore PET PRG file linker script.                                                                 
+ *                                                                                                       
+ * Produces a PRG file with a SYS command to start the program.                                          
+ */
+
+MEMORY {
+    ram (rw) : ORIGIN = 0x0401, LENGTH = 0x7BFF
+}
+
+INCLUDE commodore.ld
```

Comment:
The `commodore.ld` linker script makes the zero page from 0x02 to 0x90 available for use by the compiler. Are those regions free on the PET? (That is, the KERNAL interrupts won't clobber them, and the KERNAL is usable regardless of what is written there?)

If more/less ZP is free, it'd be a good idea to pull the zp stuff out of `commodore.ld` down into the dependent targets:
  - `cx16`
  - `pet`
  - `mega65`
  - `c64`

The `clang.cfg` in the commodore directory also affects this; it tells the compiler how much ZP the linker will later be able to satisfy.

Reply:
Unfortunately the last 3 bytes of this space don't look to be usable on the PET. 0x8D-8F are used for the jiffy clock so the KERNAL will overwrite them. There doesn't seem to be much extra free ZP apart from the odd unused byte here and there. Let me know if this means ZP stuff should be pulled down into the targets (and whether that should be implemented as part of this pull request)

Reply:
Yep, you'll need to do this as part of this PR; otherwise, the PET target could produce random behavior for large programs that use the whole ZP space. That's surprisingly common, since the compiler greedily allocates everything it can to ZP.

Reply:
Alright, looks good!